pkg/kubectl/cmd/delete.go

```diff
@@ -302,7 +302,12 @@ func DeleteResult(r *resource.Result, out io.Writer, ignoreNotFound bool, shortO
 
 		// if we're here, it means that cascade=false (not the default), so we should orphan as requested
 		orphan := true
-		return deleteResource(info, out, shortOutput, mapper, &metav1.DeleteOptions{OrphanDependents: &orphan})
+		options := &metav1.DeleteOptions{}
+		if gracePeriod >= 0 {
```

Comment:
Didn't notice that `DeleteOptions.GracePeriodSeconds` is a pointer. Then it makes a difference indeed.